Apple TV remote

Supereg opened the pull request:
This PR adds support for Apple TV Remote accessories (though without support for Siri commands).
As this PR relies on the changes made in #702 those commits are also included here. So that PR needs to be merged first. But I didn't want to wait. Although #702 is still in review, discussion can already be started for this PR.

The main API for remotes is in `src/lib/HomeKitRemoteController.ts` (with all new characteristics in `src/lib/gen/HomeKit-Remote.ts`).
For demonstration purposes I created `src/accessories/AppleTVRemote_accessory.ts`, which exposes an http api (documented in the source file) in order to play around with it.

As noted in `src/lib/HomeKitRemoteController.ts` line 126 ff. persistent storage of the target configuration is still not implemented (as I'm not quite sure how to handle plugins getting removed and added again in scenarios where for example homebridge is used). HomeKit seems to handle that fine though.

Issue regarding this topic: #612 

~ Andi

KhaosT commented on src/lib/gen/HomeKit-Remote.ts:
```diff
@@ -0,0 +1,257 @@
+
+}
+
+Characteristic.SetupDataStreamTransport = SetupDataStreamTransport;
```
Can you pull Data Stream related services/characteristics to a separate file? Since those are also being used by HomeKit Secure Video, putting it in `Remote` seems a little bit too limiting...

KhaosT commented:
And would you mind to rebase from master?

Supereg commented:
Yes rebase was the plan.
Regarding the DataStream: of course no problem. Didn’t dig into the new iOS 13 features, so didn’t know about that. Just create a new file for the data streams?
But it will take till tomorrow, going to bed now lol

KhaosT commented:
Yeah just a new file for data streams are good 🙏

Supereg commented:
rebased onto master; moved DataStream related characteristics into separate file; additionally, updated admin restriction checks in Accessory.ts (see lines 894, 1028 and 1083)

Supereg commented:
I did leave the `AudioStreamManagement` service (which is pretty generically implemented though only used for remotes currently) and the `SelectedAudioStreamConfiguration` characteristic in the `HomeKit-Remote.ts` file. The `SupportedAudioStreamConfiguration` characteristic (also exposed by IP camera) of the `AudioStreamManagement` service lies in the generated file. A bit split up but I didn't really know what the best practice is there.

KhaosT commented:
Really sorry about the long delay... was super busy with work related stuff. Will try to find a time to look it over and merge it.

Supereg commented:
Would be great 😊 Hit me up if I need to clarify anything :)

Supereg commented:
_corrected some typos and buffer allocation in latest changes_

KhaosT commented:
Thanks 🙏
